ingest_asset_opex_creator.tf

```diff
@@ -0,0 +1,32 @@
+locals {
+  ingest_asset_opex_creator_lambda_name = "${local.environment}-ingest-mapper-opex-creator"
+}
+module "ingest_asset_opex_creator_lambda" {
+  source          = "git::https://github.com/nationalarchives/da-terraform-modules//lambda"
+  function_name   = local.ingest_asset_opex_creator_lambda_name
+  handler         = "uk.gov.nationalarchives.Lambda::handleRequest"
+  timeout_seconds = 60
+  policies = {
+    "${local.ingest_asset_opex_creator_lambda_name}-policy" = templatefile("./templates/iam_policy/ingest_asset_opex_creator_policy.json.tpl", {
+      source_bucket_name      = local.ingest_raw_cache_bucket_name
+      destination_bucket_name = local.ingest_staging_cache_bucket_name
+      account_id              = var.account_number
+      lambda_name             = local.ingest_asset_opex_creator_lambda_name
+      dynamo_db_arn           = module.files_table.table_arn
+      gsi_name                = local.files_table_global_secondary_index_name
+
+    })
+  }
+  memory_size = 512
+  runtime     = "java17"
+  plaintext_env_vars = {
+    DYNAMO_TABLE_NAME  = local.files_dynamo_table_name
+    DYNAMO_GSI_NAME    = local.files_table_global_secondary_index_name
+    SOURCE_BUCKET      = local.ingest_raw_cache_bucket_name
```

Comment:
Would `SOURCE_BUCKET` be better to be passed in the event as the prefix will also be required from the event anyway?

Reply:
We're passing in batch ID and exection name to the asset-creator lambda but that's fine because the batch ID is available on the global input object, as is the execution name so we can add those to the input in the step function but the source bucket will have to be passed in from the upsert lambda which will end up looking like the json above.

Reply:
Could we not do some Step Functions stuff here with the itemSelector field, like in https://docs.aws.amazon.com/step-functions/latest/dg/concepts-asl-use-map-state-inline.html#inline-map-state-example-params ? Our state data will contain an S3 prefix for the batch, which we'd need in each invocation as an ID alone won't be enough to figure out where the object is in S3.